cmd/commands/server/command_server.go

```diff
@@ -158,14 +167,90 @@ func (cmd *Command) Start() (err error) {
 		return err
 	}
 
-	signer := cmd.createSigner(providerID)
+	providerAddress := common.HexToAddress(providerID.Address)
+	// check if node's identity is registered
+	registered, err := identityRegistry.IsRegistered(providerAddress)
+	if err != nil {
+		return err
+	}
 
+	registrationDataProvider := registry.NewRegistrationDataProvider(cmd.keystore)
+	signer := cmd.createSigner(providerID)
 	cmd.proposalAnnouncementStopped.Add(1)
-	go cmd.discoveryAnnouncementLoop(proposal, cmd.mysteriumClient, signer, stopDiscoveryAnnouncement)
+
+	// if not registered - wait indefinitely for identity registration event
+	if !registered {
+		registrationData, err := registrationDataProvider.ProvideRegistrationData(providerAddress)
+		if err != nil {
+			return err
+		}
+
+		registry.PrintRegistrationData(registrationData)
+
+		go func() {
+			killed := cmd.identityRegistrationWaitLoop(providerID, identityRegistry, stopRegistrationWaitLoop)
+			if killed {
+				cmd.proposalAnnouncementStopped.Done()
+				return
+			}
+			cmd.discoveryAnnouncementLoop(proposal, cmd.mysteriumClient, signer, stopDiscoveryAnnouncement)
+		}()
+	} else {
+		go cmd.discoveryAnnouncementLoop(proposal, cmd.mysteriumClient, signer, stopDiscoveryAnnouncement)
+	}
+
+	err = cmd.registerTequilAPI(identityRegistry, providerID, registrationDataProvider)
+	if err != nil {
+		return err
+	}
 
 	return nil
 }
 
+func (cmd *Command) registerTequilAPI(statusProvider registry.IdentityRegistry, providerID identity.Identity, registrationDataProvider registry.RegistrationDataProvider) error {
```

Comment:
startServingTequilAPI ?

Reply:
Also this function does a bit too much - registers endpoints on routers and starts http server. It could be split into two methods - one for endpoints registering, another for starting http

Reply:
Dont need this function at all #324 implements Tequilapi on VPN server

Reply:
removed, now node contains only former mysterium_client tequilapi. 
Removed /identity/current endpoint